### ISSUE TITLE
Automatically clear tabs/spaces from empty lines?

### ISSUE THREAD

**Issue by dillonfagan** [elementary/code#581]
Feature request: Like Visual Studio, it would be awesome if Code could satisfy my OCD by clearing spacing (and tabs) from empty lines, i.e. lines with no text in them. The company I work for has made it part of their coding standards that blank lines should be completely blank, including any excess spacing.

**Comment by dillonfagan** [comment]
Ah now I feel slightly foolish... except am I right that this only works when manually saving? I'd like for trailing spaces to be stripped in real-time if that's possible.

**Comment by arshubham** [comment]
yeah you are right. Title can me made a little clearer. Anyway someone needs to just call it on enter key press event instead of save event and submit a new plugin. Is this how vscode handles it? Does it remove the whitespaces from whole file or just the current line?

**Comment by dillonfagan** [comment]
I think as you type, Visual Studio (not sure about VS Code) removes trailing spaces from the current line after hitting enter/return.

**Comment by jeremypw** [comment]
If stripping is enabled for automatic save then it should not be applied to the line being typed.

**Comment by jeremypw** [comment]
This could be annoying as I often temporarily move away from the current line which might contain space (tabs) which I do not want removed at that point as I want to paste something afterwards.  You can force stripping just by pressing `<Ctrl>S` if you want.

**Comment by jeremypw** [comment]
I must admit I find it irritating to have to manually save in order to remove extraneous white space even when autosave is enabled.
